src/App.tsx

```diff
@@ -1,57 +1,64 @@
-  const { name, born, died } = peopleFromServer[0];
-
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
-
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
+const preparedPeople = (
+  people: Person[],
+  query: string,
+) => {
+  const peopleCopy = [...people].filter(person => (
+    person.name.toLowerCase().includes(query.toLowerCase())
+  ));
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
+  return peopleCopy;
+};
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
+export const App: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+  const people = useMemo(() => (
+    preparedPeople(peopleFromServer, appliedQuery)),
+  [appliedQuery]);
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
+  const handleQueryChange = (name: string) => {
+    setQuery(name);
+    applyQuery(name);
+  };
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
+  return (
+    <main className="section">
+      <h1 className={classNames('title', {
+        'has-text-link': selectedUser?.sex === 'm',
+        'has-text-danger': selectedUser?.sex === 'f',
+      })}
+      >
+        {selectedUser ? (
+          `${selectedUser.name} (${selectedUser.born} = ${selectedUser.died})`
+        ) : (
+          'No selected person'
+        )}
+      </h1>
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
-      </div>
+      <Dropdown
+        people={people}
+        query={query}
+        selectedUser={selectedUser}
+        setQuery={setQuery}
+        setSelectedUser={setSelectedUser}
+        handleQueryChange={handleQueryChange}
+        appliedQuery={appliedQuery}
+        setAppliedQuery={setAppliedQuery}
+      />
```

Comment:
I think it would be better to leave the input in the App component to avoid passing so many props.